### PR TITLE
Managing ssl parameter into query for JetPack.

### LINKED ISSUE #11249: Media: Media may not load with Jetpack connection even when it works on other clients
### Expected behavior

In the Media section, all the site's media should appear and you should be able to preview it when selecting any media item.

### Actual behavior

A user reported that they can’t see their media in the Android app only when logged in with their WordPress.com account. All files give the error: “Media could not be found”

These are the place they CAN see media:

- in the iOS app
- at /wp-admin/uploads on the same device
- at WordPress.com/media on the same device
- in the app on the same device when they log in by site address

What the user has tried:

- Turning off security plugins
- Uninstalling/reinstalling the app
- Logging in and out again
- Whitelisting Jetpack’s IP ranges in `fail2ban`

App logs suggest that the media is fetched without any issues:

```
67 - [Dec-27 09:04 API] Dispatching action: MediaAction-FETCH_MEDIA_LIST
68 - [Dec-27 09:04 API] Dispatching action: SiteAction-FETCH_SITE
69 - [Dec-27 09:04 API] Dispatching action: SiteAction-UPDATE_SITE
70 - [Dec-27 09:04 DB] Site found by (local) ID: 1
71 - [Dec-27 09:04 DB] Updating site: [REDACTED]
72 - [Dec-27 09:04 MEDIA] Fetched media list for site with size: 48
73 - [Dec-27 09:04 API] Dispatching action: MediaAction-FETCHED_MEDIA_LIST
```

Given that their media library loads as expected over XML-RPC (when they are logged in to the app via the site address) and it loads as expected on other clients (iOS, Calypso) via the Jetpack connection / when logged in to a WordPress.com account, I would expect it to work that way in the Android app.

We haven't been able to reproduce this issue on another site, but this merits investigation because it seems like the Android app must be handling the media for Jetpack sites differently from other clients, in a way that makes this issue possible.

User report: 2550066-zen
Discussion: p4a5px-2u0-p2

### PR DESCRIPTION
**Pull request by develric**
Fixes #11249

Some more context in the inked issue above and in `p1584956107248200-slack-ios-osx-dev`

**NOTE:  I do not have full clarity about the correct procedure to final get merged this PR into the original WP-Utils subtree (after review/approve, let's sync before to merge this one).**

My best understanding is the steps reported here `p1570688351009600-slack-android-dev` but the tests update related to the PR here #11524 AFAIU does not seem to be merged into WPUtils develop (cc @jkmassel in case I'm missing something if he can help to clarify 🙇‍♂️)

### To test:
- Check you can access Media gallery images without errors with:
    - WP.com site, 
    - https JetPack site (added by address or by WP.com login), 
    - http JetPack site (added by address or by WP.com login) 
- Check the WP-Utils connected tests pass
- Check you can repro and get fixed the issue the user had in #11249 setting up Charles Proxy with SSP (further details here `p99K0U-ZF-p2`)

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by oguzkocer** [comment]
This issue is targeting the current frozen milestone which is about to be released. I am bumping it to `14.6` which is when a PR that lands to `develop` in the next 2 weeks will be released. If it's more urgent than that, please make sure to target `release/14.5` branch which will be available later today and update the milestone for the issue accordingly.

File: libs/utils/WordPressUtils/src/androidTest/java/org/wordpress/android/util/PhotonUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class PhotonUtilsTest {
+    @Test
+    public void getPhotonImageUrlIsEmptyWhenUrlIsNull() {
+        String photonUrl = PhotonUtils.getPhotonImageUrl(null, 0, 1);
+
+        assertEquals("", photonUrl);
```
Comment (planarvoid) [inline libs/utils/WordPressUtils/src/androidTest/java/org/wordpress/android/util/PhotonUtilsTest.java]: I'd rather use `assertThat(photonUrl).isEmpty()` and generally use `assertThat` than other asserts. I think it's more readable and I like the consistency.

Comment (develric) [reply libs/utils/WordPressUtils/src/androidTest/java/org/wordpress/android/util/PhotonUtilsTest.java]: Agree. AFAIU in the WPUtils we use JUnit and Hamcrest, so I moved using that assertThat pattern here 9881777ae35ad408285b5452a5639248e79b543d 😊

**Comment by develric** [comment]
Hey @planarvoid , thanks for the review! I moved to the assertThat pattern here 9881777ae35ad408285b5452a5639248e79b543d and also changed the lib version here 12b7277b44b8d996bfacaefbf91071b99f22fa16 (for what I now understand I get we need to manage it here as well as in the subtree).

**Comment by develric** [comment]
Hey @planarvoid, sorry to ping again on this one 😊! As per an alignmentt chat with @oguz, we decided to keep the version of the Utils Lib here since we are not making a release from this right now (there are some other PRs modifications sync to do also and the release will be managed in a separate task).

**Pull request merged**